test/Microsoft.AspNetCore.Mvc.Core.Test/RedirectToActionResultTest.cs

```diff
@@ -83,6 +83,41 @@ public class RedirectToActionResultTest
                 "No route matches the supplied values.");
         }
 
+        [Fact]
+        public async Task RedirectToAction_Execute_WithFragment_PassesCorrectValuesToRedirect()
+        {
+            // Arrange
+            var expectedUrl = "/Home/SampleAction#test";
+            var expectedPermanentFlag = false;
+
+            var httpContext = new Mock<HttpContext>();
+            httpContext
+                .SetupGet(o => o.RequestServices)
+                .Returns(CreateServices().BuildServiceProvider());
+
+            var httpResponse = new Mock<HttpResponse>();
+            httpContext
+                .Setup(o => o.Response)
+                .Returns(httpResponse.Object);
```

Comment:
Guess the existing tests weren't the best model to follow. Don't change them but do avoid unnecessary mocks in tests you add.